lib/manageiq/util/file_splitter.rb

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env ruby
+#
+# Yes, you guessed it... this is basically the `split` command...
+#
+# The intent of this is to allow for greater flexibility and utility than is
+# provided by the BSD/GNU variants however, specifically allowing to pipe from
+# a pg_dump directly to a upload target (whether it be a mounted volume or
+# something like a FTP endpoint, though the former is basically already
+# supported with vanilla `split`).
+#
+# FTP, specifically, will be supported natively since ruby has Net::FTP support
+# built in (feature WIP).  This also allows for this to work correctly cross
+# platform, without having to be concerned with differences in `split`
+# functionality.
+
+require_relative 'ftp_lib.rb'
+require 'optparse'
+
+class FileSplitter
+  include ManageIQ::Util::FtpLib
+
+  attr_accessor :input_file, :byte_count
+
+  class << self
+    attr_writer :instance_logger
+
+    # Don't log by default, but allow this to work with FtpLib logging.
+    def instance_logger
+      @instance_logger ||= Logger.new(File::NULL)
+    end
+  end
+
+  def self.run(options = nil)
+    options ||= parse_argv
+    new(options).split
+  end
+
+  def self.parse_argv
+    options = {}
+    OptionParser.new do |opt|
+      opt.on("-b", "--byte-count=BYTES", "Number of bytes for each split") do |bytes|
+        options[:byte_count] = parse_byte_value(bytes)
+      end
+      opt.on("--ftp-host=HOST", "Host of the FTP server") do |host|
+        options[:ftp_host] = host
+      end
+      opt.on("--ftp-dir=DIR", "Dir on the FTP server to save files") do |dir|
+        options[:ftp_dir] = dir
+      end
+      opt.on("-v", "--verbose", "Turn on logging") do
+        options[:verbose] = logging
+      end
+    end.parse!
+
+    input_file, file_pattern = determine_input_file_and_file_pattern
+
+    options[:input_file]     = input_file
+    options[:input_filename] = file_pattern
+
+    options
+  end
+
+  def initialize(options = {})
+    @input_file     = options[:input_file] || ARGF
+    @input_filename = options[:input_filename]
+    @byte_count     = options[:byte_count] || 10_485_760
```

Comment:
Prefer `10.megabytes` for readability (or if not ActiveSupport, `10 * 1024 * 1024 # 10.megabytes`)